addons/event_sale/tests/test_event_sale.py

```diff
@@ -473,3 +473,35 @@ def test_cancel_so(self):
         self.sale_order._action_cancel()
         self.assertEqual(len(event.registration_ids), 1)
         self.assertEqual(event.registration_ids.state, 'cancel')
+
+    @users('user_sales_salesman')
+    def test_adjusted_quantity_in_sale_order(self):
+        # Initial setup: Creating a sale order with 3 tickets
+        customer_so = self.customer_so.with_user(self.env.user)
+        ticket = self.event_0.event_ticket_ids[0]
+
+        customer_so.write({
+            'order_line': [
+                (0, 0, {
+                    'event_id': self.event_0.id,
+                    'event_ticket_id': ticket.id,
+                    'product_id': ticket.product_id.id,
+                    'product_uom_qty': 3,
+                    'price_unit': 10,
+                })
+            ]
+        })
+
+        editor = self.env['registration.editor'].with_context({'default_sale_order_id': customer_so.id}).create({})
+        editor.action_make_registration()
+
+        registration_to_cancel = self.event_0.registration_ids[0]
+        registration_to_cancel.action_cancel()
+
+        registrations = self.env['event.registration'].search([('sale_order_id', '=', customer_so.id)])
+        registrations._compute_registration_status()
```

Comment:
Seems like a symptom of a bad depends if the compute needs to be called manually. Can we avoid that?